src/aobasis/ao_util.F

```diff
@@ -281,10 +279,16 @@ FUNCTION exp_radius(l, alpha, threshold, prefactor, epsin) RESULT(radius)
          CPABORT("The requested threshold is zero")
       END IF
 
+      IF (PRESENT(epsin)) THEN
+         epsiter = epsin
+      ELSE ! arbitrary scaled epsilon?
+         epsiter = EPSILON(epsiter)*(1.0E-12_dp/2.22044604925031E-16_dp)
```

Comment:
isn't this equivalent to simply `epsiter = 1.0E-12_dp` ?

At least `2.22044604925031E-16_dp` is the epsilon of our `dp` type and `gcc -O0` optimizes this away to `epsiter = 1.0E-12_dp`, see https://gcc.godbolt.org/z/65G8Tx

Reply:
maybe @vondele knows more about this? Based on git history, this seems to be a remnant of the single-precision experiments

Reply:
I believe it is a hand-calibrated eps to ensure the algorithm terminates (at least for the test cases). The problem with the old/current implementation is it may not terminate for too small espilon. This is resolved.

Reply:
btw, can you recommend a test for eri_mme_lattice_summation? It currently uses exp_radius although exp_radius_fast is already a dependency of eri_mme_lattice_summation.

Reply:
I think @dev-zero is right, this was needed for single precision experiments. This stuff is from long ago, so I've forgotten the details. These routines need to be handled with care, i.e. are both critical for performance and accuracy. Underestimating the radius leads to large inaccuracies elsewhere, significantly overestimating leads to larger costs. The target accuracy should be on the order of a grid spacing (IIRC). Nevertheless, these routines are old and might definitely benefit from a new round of TLC.

Reply:
> btw, can you recommend a test for eri_mme_lattice_summation? It currently uses exp_radius although exp_radius_fast is already a dependency of eri_mme_lattice_summation.

It turns out that in the current implementation, `exp_radius_fast` is not used at all and I don't remember whether this is what I intended in pr #369. I will do some testing on my own to find out if it matters whether `exp_radius` or `exp_radius_fast` is used.

Reply:
@pseewald Thank you!  Please pull this latest PR into your test branch (I moved exp_radius_fast into ao_util.F because of using it to implement exp_radius; see commit log about cross-dependency).

With your insight we should decide if we want to take the new exp_radius, or if we only want to take exp_radius_fast. I also think with own insight and Joost's input, we may want to adjust tolerances for ccurrently failing tests (since no current/later reference might be "better" or more correct). With this exp_radius in general, it is hard to reproduce the exact same result when the implementation changed (even slightly). Please note, tests also/even fail with a much tighter iteration interval (so reference values are somewhat arbitrary).

Regarding performance, `tests/LIBTEST/test_eri_mme_performance.inp` is not dominated by exp_radius but other ERI related pieces such as error control (I will detail this later).

Reply:
@hfp I know for sure that the error control overhead in `test_eri_mme_performance.inp` is not representative for performance in actual applications since it's only called once. Other than that, any insights are very welcome! Some time ago I identified the `pgf_sum_3c_*` routines as main bottlenecks and optimized them in #369.

Reply:
@hfp I finally checked that indeed `exp_radius_fast` is no longer needed for me. So I plan to remove it from the `eri_mme_gaussian` module in a later commit.

Reply:
@pseewald That's also part of this PR.

I think I could update the tolerances. However, I do not know how to avoid the "failure" with `test_eri_mme_accuracy.inp`. It brings up this ultimate argument `Actual error greater than upper bound estimate`.

Reply:
note that you should not update the tolerance, but the reference values for this kind of test. Tolerances are used to take into account variation due to machine architectures, threading etc, the tolerance on most tests is far stricter than all the thresholds in the code actually implies. If one changes the algorithms (like here) reference values need to be updated.

Reply:
@vondele Thank you for the comment. I will do so if this PR goes forward.